### DISCUSSION TITLE
I am trying to build a golang project which refers https://github.com/openconfig/gnmi

### DISCUSSION THREAD

**Discussion by ChiragR7** [bazelbuild/bazel#23871] [Q&A]
It generated BUILD.bazel file as

```
load("@io_bazel_rules_go//go:def.bzl", "go_library")

go_library(
    name = “test”,
    srcs = [
        “test.go”,
    ],
    importpath = “git-url-here”,
    visibility = ["//visibility:public"],
    deps = [
        "@com_github_openconfig_gnmi//proto/gnmi",
        "@org_golang_google_protobuf//proto",
    ],
)
```

and while running 

`bazel build //... `

It gave error 

`ERROR: error loading package '@@com_github_openconfig_gnmi//proto/gnmi': Unable to find package for @@com_github_grpc_grpc//bazel:cc_grpc_library.bzl: The repository '@@com_github_grpc_grpc' could not be resolved: Repository '@@com_github_grpc_grpc' is not defined.`

tried fixed it by adding 

```
http_archive(
    name = "com_github_grpc_grpc",
    strip_prefix = "grpc-1.66.0",  # Adjust the prefix based on the version you are using
    urls = ["https://github.com/grpc/grpc/archive/refs/tags/v1.66.0.zip"],  # Use the latest version you need
)
```

It started giving another error 

```
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
compilepkg: missing strict dependencies:
        /private/var/tmp/_bazel_usr0b903048f40c7721c45e60c73f47d067/sandbox/darwin-sandbox/1482/execroot/_main/services/test.go: import of "github.com/openconfig/gnmi/proto/gnmi"
No dependencies were provided.
```

how to move forward with this ?

**Comment by sluongng** [comment]
This sounds like an issue specific to Gazelle's go_repository generating BUILD files incorrectly.
If you could create a minimal reproduction of the issue, do report it to https://github.com/bazelbuild/bazel-gazelle/issues
